editor/plugins/canvas_item_editor_plugin.cpp

```diff
@@ -4685,24 +4684,16 @@ void CanvasItemEditor::_focus_selection(int p_op) {
 		}
 	};
 
-	if (p_op == VIEW_CENTER_TO_SELECTION) {
-		center = rect.get_center();
-		Vector2 offset = viewport->get_size() / 2 - EditorNode::get_singleton()->get_scene_root()->get_global_canvas_transform().xform(center);
-		view_offset -= (offset / zoom).round();
-		update_viewport();
-
-	} else { // VIEW_FRAME_TO_SELECTION
-
-		if (rect.size.x > CMP_EPSILON && rect.size.y > CMP_EPSILON) {
-			real_t scale_x = viewport->get_size().x / rect.size.x;
-			real_t scale_y = viewport->get_size().y / rect.size.y;
-			zoom = scale_x < scale_y ? scale_x : scale_y;
-			zoom *= 0.90;
-			viewport->queue_redraw();
-			zoom_widget->set_zoom(zoom);
-			call_deferred(SNAME("_popup_callback"), VIEW_CENTER_TO_SELECTION);
-		}
+	if (p_op == VIEW_FRAME_TO_SELECTION && rect.size.x > CMP_EPSILON && rect.size.y > CMP_EPSILON) {
+		real_t scale_x = viewport->get_size().x / rect.size.x;
+		real_t scale_y = viewport->get_size().y / rect.size.y;
+		zoom = scale_x < scale_y ? scale_x : scale_y;
+		zoom *= 0.90;
+		zoom_widget->set_zoom(zoom);
+		viewport->queue_redraw();
 	}
+	// Redrawing in VIEW_FRAME_TO_SELECTION is queued. The global canvas transform must update first, so defer the centering.
+	call_deferred(SNAME("center_at"), rect.get_center());
```

Comment:
So now it's also deferred for `VIEW_CENTER_TO_SELECTION` even though deferring is not needed in that case? :thinking:
If that's the case then maybe it would make more sense to defer only from the `if` block above and add a non-deferring `else`?

Reply:
I don't see the point but sure.